### ISSUE TITLE
AccessTokenPlugin AuthorizationType 

### ISSUE THREAD

**Issue by ranhsd** [Moya/Moya#1760]
Hi, 
I noticed that the access token plugin authroization types provide the following options: 

```
public enum AuthorizationType: String {
    /// No header.
    case none

    /// The `"Basic"` header.
    case basic = "Basic"

    /// The `"Bearer"` header.
    case bearer = "Bearer"
}
```

Now, there are some backends which expect only a JWT token in the autorization header without the Bearer or Basic prefixes. 

[FeathersJS](https://docs.feathersjs.com/) is a good example for such backend. There you need to provide only (and only) the JWT token in the Autorization header in order to authenticate yourself with the backend.
Also, there are some API's which implemented custom prefix for their Authroziation header like [Shippo API](https://goshippo.com/docs/auth/) there they expect that users will provide a **ShippoToken**  prefix to the token to autorize themselves. 

Base on the info above I think that Moya should introduce a new value to the **AuthorizationType** enum of **custom**  (or something with better name :)). This value should get a string that will be concatenated to the token before each request which requires it. 

So the new enum will be: 

```
public enum AuthorizationType: String {
    /// No header.
    case none
    
    /// The `"Basic"` header.
    case basic = "Basic"
    
    /// The `"Bearer"` header.
    case bearer = "Bearer"
    
    case custom(prefix: String?)
}

```

Notice that prefix is optional so if the user send nil it will use the token itself without any prefix and will use the token as it is. If not it will concatenate the prefix with the token and will add a space to separate between them.

Thanks!

**Comment by SD10** [comment]
Hey @ranhsd, this feature was added by #1611. It's currently on Moya development. Will be available in Moya 12.0.0

**Comment by ranhsd** [comment]
Hi @SD10 wow it's amazing. The implementation is exactly what I wanted. They even use the same name :) 
BTW - it's the first time I see it. 

Thanks!

**Comment by SD10** [comment]
This is now available in Moya 12.0 👍
